homeassistant/components/twinkly/__init__.py

```diff
@@ -33,9 +33,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except (asyncio.TimeoutError, ClientError) as exception:
         raise ConfigEntryNotReady from exception
 
+    software_version = await client.get_firmware_version()
```

Comment:
Should this happen inside the try in case it times out or has a transient error?

Reply:
I added it to the same try as device_info.  Not 100% sure about that, as device_info could theoretically succeed while software_version fails.  But at the same time they should both succeed if the device is ready.